### PR TITLE
API: Deprecate old incremental append scans

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR deprecates old incremental append-only scans in favor of the new API.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
@stevenzwu @rdblue, could you take a look?

File: api/src/main/java/org/apache/iceberg/TableScan.java [hunk 1]
```diff
@@ -69,7 +69,10 @@ default TableScan select(String... columns) {
    * @param toSnapshotId read append data up to this snapshot id
    * @return a table scan which can read append data from {@code fromSnapshotId} exclusive and up to
    *     {@code toSnapshotId} inclusive
+   * @deprecated since 1.0.0, will be removed in 1.1.0; use {@link Table#newIncrementalAppendScan()}
```
Comment (stevenzwu) [inline api/src/main/java/org/apache/iceberg/TableScan.java:72]: I assume we decided that the next major/minor release will be 1.0.0. there won't be a 0.15.0.

**Review by stevenzwu — APPROVED** [review]

File: api/src/main/java/org/apache/iceberg/TableScan.java (hunk 1 above)
Comment (aokolnychyi) [reply api/src/main/java/org/apache/iceberg/TableScan.java:72]: Yeah, that seems to be the assumption. @rdblue, could you confirm?

Comment (rdblue) [reply api/src/main/java/org/apache/iceberg/TableScan.java:72]: Yes, this should be removed in 2.0.0.

Comment (aokolnychyi) [reply api/src/main/java/org/apache/iceberg/TableScan.java]: Fixed.

**Pull request merged**